docs/snapshot-restore.md

```diff
@@ -0,0 +1,134 @@
+3. Wait for the restore to complete. Searches will continue to use the old
+index.
+3. Switch the alias: `mainstream` -> `restored-mainstream-2016-02-25t17:11:04z-6d2a7dc4-a3b7-4572-b388-c59dc2c40be5`
+4. Searches now use the new index.
+
+## Repository setup
+
+If you have an S3 bucket set up, you can create a snapshot repository yourself
+by running:
+
+```
+rake rummager:snapshot:create_repository[repository_name]
+```
+
+This assumes you have some environment variables available:
+
+- AWS_ACCESS_KEY_ID
+- AWS_ACCESS_SECRET_ACCESS_KEY
+
+The name of the bucket to be used is configured in `elasticsearch.yml`.
+
+If you want to create your own repository (e.g. for development), you might
+want to create a filesystem backed repository instead. This is documented in
+the [elasticsearch documentation for snapshot restore](https://www.elastic.co/guide/en/elasticsearch/reference/1.4/modules-snapshots.html).
+
+You won't be able to use the `rummager:snapshot:list` and `rummager:snapshot:latest`
+tasks without an S3 bucket, but the basic snapshot/restore can work with any
+repository type.
+
+## Start a snapshot
+Normally, snapshots are triggered regularly from a cron job.
+
+You can initiate a snapshot manually by running:
+
+```shell
+rake rummager:snapshot:run[repository_name,snapshot_name]
+```
+
+If snapshot_name is left out, Rummager generates one based on the current time.
+
+If the repository does not exist, or elasticsearch is unable to process the
+request, the task exits with a non zero code and prints a stack trace.
+
+Otherwise the exit code will be zero.
+
+## Monitoring snapshots
+### Check if a snapshot has completed yet
+```shell
+rake rummager:snapshot:check[repository_name,snapshot_name]
+```
+This returns the status of a snapshot operation, e.g. "SUCCESS".
+
+### Find the last successful snapshot
+```shell
+# Print the name of the latest snapshot
+rake rummager:snapshot:latest[repository_name]
+
+# Print the last snapshot excluding any after Jan 1st 2016
+rake 'rummager:snapshot:latest[repository_name, 2016-01-01 00:00:00]
+```
+
+### List all snapshots in the S3 bucket
+```shell
+# Includes in-progress snapshots
+rake rummager:snapshot:list[repository]
+```
+
+## Restore the latest snapshot
+```shell
+INDEX_NAMES=all rake rummager:snapshot:restore[repository_name,snapshot_name]
+```
+Prints the names of the indexes that will be created.
+
+Exits with a non-zero code and prints a stack trace if elasticsearch is
+unable to handle a request. Only one restore operation can happen at a time.
+
+## Monitor a restore from snapshot
+```shell
+rake rummager:check_recovery[index_name]
+```
+Prints "true" or "false", depending on whether the restore is complete.
+This uses the elasticsearch recovery API.
+
+## Switch an index alias to a restored index
+
+Once an index has been successfully restored from a snapshot, you can switch
+over the alias:
+```shell
+INDEX_NAMES=mainstream rake rummager:switch_to_named_index[new_index_name]
```

Comment:
It comes from the previous restore task. Added a sentence above to clarify this.